juju/juju
# Add remote firewaller facade

## Description of change

This PR adds a new remote firewaller facade. The facade will be used by the firewaller worker and provides firewall/network related functionality against a different (remote) model, where a cross model application is hosted and a relation established.

The only method currently is WatchSubnets() which notifies of changes to subnets in a remote model. The next PR will move the IngressSubnetsForRelation() method off the remote relation facade and onto this new remote firewaller facade.

The worker uses a helper method to construct a remote firewaller facade to access a given model - the remote relations worker uses the same methodology so some code to manage the underlying API connection itself was extracted to a helper. At the moment, the result is a facade pointing to a model on the same controller, but in future will be used to allow cross controller interactions transparent to the worker.

The PR includes all the components of the facade - api, apiserver, state functionality. The firewaller worker manifold was restructured to allow tests to be updated later to plug in mocks etc (as part of the work to plug in the remote firewaller facade).

## QA steps

bootstrap aws
create 2 models and deploy an app to each
relate the apps across models
expose the app
check the ports
